tests/TestCase/View/Helper/FlashHelperTest.php

```diff
@@ -154,4 +154,19 @@ public function testFlashWithPluginElement()
         $expected = 'this is the plugin element';
         $this->assertEquals($expected, $result);
     }
+
+    /**
+     * test that when View theme is set, flash element from that theme (plugin) is used.
+     *
+     * @return void
+     */
+    public function testFlashWithTheme()
+    {
+        Plugin::load('TestTheme');
+
+        $this->View->theme = 'TestTheme';
+        $result = $this->Flash->render('flash');
+        $expected = 'flash element from TestTheme';
+        $this->assertContains($expected, $result);
```

Comment:
The changes in this PR don't do anything - This test will pass using 3.0's flash helper.

```
www-data @ dev [ /home/andy/repos/cakephp/cakephp ] (issue-5920 $=)
-> git status
# On branch issue-5920
nothing to commit (working directory clean)
www-data @ dev [ /home/andy/repos/cakephp/cakephp ] (issue-5920 $=)
-> phpunit tests/TestCase/View/Helper/FlashHelperTest.php 
PHPUnit 4.3.4 by Sebastian Bergmann.

Configuration read from /home/andy/repos/cakephp/cakephp/phpunit.xml.dist

.....

Time: 106 ms, Memory: 6.75Mb

OK (5 tests, 8 assertions)
www-data @ dev [ /home/andy/repos/cakephp/cakephp ] (issue-5920 $=)
-> git checkout origin/3.0 src/
www-data @ dev [ /home/andy/repos/cakephp/cakephp ] (issue-5920 +$=)
-> git diff --cached
diff --git src/View/Helper/FlashHelper.php src/View/Helper/FlashHelper.php
index 7fd83ef..f871c94 100644
--- src/View/Helper/FlashHelper.php
+++ src/View/Helper/FlashHelper.php
@@ -80,11 +80,7 @@ class FlashHelper extends Helper
         $flash = $options + $flash;
         $this->request->session()->delete("Flash.$key");
 
-        $element = $flash['element'];
-        if ($this->_View->theme && strpos($element, '.') === false) {
-            $element = $this->_View->theme . '.' . $element;
-        }
-        return $this->_View->element($element, $flash);
+        return $this->_View->element($flash['element'], $flash);
     }
 
     /**
www-data @ dev [ /home/andy/repos/cakephp/cakephp ] (issue-5920 +$=)
-> phpunit tests/TestCase/View/Helper/FlashHelperTest.php 
PHPUnit 4.3.4 by Sebastian Bergmann.

Configuration read from /home/andy/repos/cakephp/cakephp/phpunit.xml.dist

.....

Time: 109 ms, Memory: 7.00Mb

OK (5 tests, 8 assertions)
www-data @ dev [ /home/andy/repos/cakephp/cakephp ] (issue-5920 +$=)
-> 
```